modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/150_tsdb.yml

```diff
@@ -191,3 +191,245 @@ index without timestamp with pipeline:
+          - '{ "create": { "dynamic_templates": { "data": "counter_metric" } } }'
+          - '{ "@timestamp": "2023-09-01T13:03:10.138Z","data": "30", "attributes.dim": "B", "resource_attributes.dim": "D" }'
+          - '{ "create": { "dynamic_templates": { "data": "counter_metric" } } }'
+          - '{ "@timestamp": "2023-09-01T13:03:10.238Z","data": "40", "attributes.dim": "B", "resource_attributes.dim": "D" }'
+
+  - do:
+      search:
+        index: k9s
+        body:
+          size: 0
+
+  - match: { hits.total.value: 4 }
+
+  - do:
+      search:
+        index: k9s
+        body:
+          size: 0
+          aggs:
+            filterA:
+              filter:
+                term:
+                  dim: "C"
+              aggs:
+                tsids:
+                  terms:
+                    field: _tsid
+
+  - length: { aggregations.filterA.tsids.buckets: 1 }
+  - match: { aggregations.filterA.tsids.buckets.0.key: { "resource_attributes.dim": "C", "attributes.dim": "A" } }
+  - match: { aggregations.filterA.tsids.buckets.0.doc_count: 2 }
+
+  - do:
+      search:
+        index: k9s
+        body:
+          size: 0
+          aggs:
+            filterA:
+              filter:
+                term:
+                  attributes.dim: A
+              aggs:
+                tsids:
+                  terms:
+                    field: _tsid
+
+  - length: { aggregations.filterA.tsids.buckets: 1 }
+  - match: { aggregations.filterA.tsids.buckets.0.key: { "resource_attributes.dim": "C", "attributes.dim": "A" } }
+  - match: { aggregations.filterA.tsids.buckets.0.doc_count: 2 }
+
+---
+dynamic templates - subobject in passthrough object error:
+  - skip:
+      version: " - 8.12.99"
+      features: "default_shards"
+      reason: "Support for dynamic fields was added in 8.13"
+  - do:
+      catch: /Mapping definition for \[attributes\] has unsupported parameters\:\ \ \[subcategory \:\ \{attributes=\{dim=\{type=keyword\}\}\}\]/
+      indices.put_index_template:
+        name: my-dynamic-template
+        body:
+          index_patterns: [k9s*]
+          data_stream: {}
+          template:
+            settings:
+              index:
+                mode: time_series
+
+            mappings:
+              properties:
+                attributes:
+                  type: passthrough
+                  dynamic: true
+                  time_series_dimension: true
+                  subcategory:
+                    attributes:
+                      dim:
+                        type: keyword
+
+---
+dynamic templates - passthrough not under root error:
+  - skip:
+      version: " - 8.12.99"
+      features: "default_shards"
+      reason: "Support for dynamic fields was added in 8.13"
+  - do:
+      catch: /Tried to add passthrough subobject \[attributes\] to object \[resource\], passthrough is not supported as a subobject/
```

Comment:
Why can't object types have a property of type passthrough? I think we were discussing the opposite where properties of the passthrough field type can't be other objects i.e. `subobjects` would be implicitly set to `false` for passthrough field types.

Reply:
It complicates how aliases are added to the root object. 

We can look into this in a follow-up change - unless Martijn knows of a simple way to navigate to the root object from the passthrough object builders.

Reply:
Gotcha. Another alternative would be to recurse the object mappers.

Reply:
Yeah I'll prioritize supporting this and non-keyword dimension fields.